app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: [:edit, :update, :destroy]
+  before_action :check_permission, only: [:edit, :destroy]
+
+  def new
+    @comment=current_post.comments.build
+  end
+
+  def create
+    @comment=current_post.comments.build(comment_params)
+    @comment.user_id=current_user.id
+
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to current_post, notice: 'Comment was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def edit; end
```

Comment:
This is not needed. The updates will be done through an ajax request (PUT /comment/:id)